rke/templates/metrics_v0.5.0.go

```diff
@@ -213,6 +213,7 @@ spec:
           readOnlyRootFilesystem: true
           runAsNonRoot: true
           runAsUser: 1000
+          allowPrivilegeEscalation: false
```

Comment:
We need to introduce a new template as per discussions in this thread - https://github.com/rancher/kontainer-driver-metadata/pull/938/files#r953052960

Reply:
Alright! Few questions. The next template verion should be `metrics_v0.5.1.go` ? And what K8s version should support this new metrics version? The current version has `">=1.20.14-rancher2-1 <1.23.3-rancher1-1": metricsServerv050` see https://github.com/rancher/kontainer-driver-metadata/blob/dev-v2.7/rke/templates/templates.go#L160
cc: @kinarashah